Change some typedef structs into normal structs

syyyr opened the pull request:
Hi, while working on my C++ wrapper for sysrepo, I found out that some of the structs are defined as typedefs. This means that you can't forward declare them. I think this change should not have an effect on existing code, but I'm not sure. What do you think?

I'm creating this as a draft, because I'm not sure what structs I'll have to change.

syyyr commented:
Ideally, I would like all structs to be in the C "struct namespace" and then possibly in the global "typedef" namespace.

michalvasko commented on src/sysrepo_types.h:
```diff
@@ -244,8 +244,9 @@ typedef struct {
 
     /** Data of an element (if applicable), properly set according to the type. */
     sr_data_t data;
+} ;
```
Extra space.

syyyr replied:
done

michalvasko commented:
I have no problem with the change and any like this. However, please rebase your branch to the current `devel`, you seem to have quite an old revision.

syyyr commented:
>I have no problem with the change and any like this.

Great! I changed all (or most) of them.

>However, please rebase your branch to the current devel, you seem to have quite an old revision.

The patch is now rebased to devel.
